fuse_constraints/include/fuse_constraints/normal_prior_orientation_3d_euler_cost_functor.hpp

```diff
@@ -133,19 +153,22 @@ class NormalPriorOrientation3DEulerCostFunctor
                     "I should probably be more informative here");
           }
       }
-      residuals[i] = angle - T(b_[i]);
     }
 
-    Eigen::Map<Eigen::Matrix<T, Eigen::Dynamic, 1>> residuals_map(residuals, A_.rows());
+    residuals[0] = angle[0];
+    residuals[1] = angle[1];
+    residuals[2] = angle[2];
+
+    Eigen::Map<Eigen::Matrix<T, 3, 1>> residuals_map(residuals, A_.rows());
     residuals_map.applyOnTheLeft(A_.template cast<T>());
 
     return true;
   }
 
 private:
-  fuse_core::MatrixXd A_;  //!< The residual weighting matrix, most likely the square root
+  fuse_core::Matrix3d A_;  //!< The residual weighting matrix, most likely the square root
                            //!< information matrix
-  fuse_core::VectorXd b_;  //!< The measured 3D orientation (quaternion) value
+  fuse_core::Vector4d b_;  //!< The measured 3D orientation (quaternion) value
```

Comment:
Switching to static size here is needed? Is this a standalone optimization that can also be applied upstream?

Reply:
My changes in this file came from tests that I tried some time ago and are not valid. I will revert it to the previous commit